Запрос на слияние ветки develop в ветку main

Manikgog opened the pull request.

Manikgog committed:
Написан класс UserController и заготовка класса UserService.

Manikgog committed:
Продолжение написания класса UserCon

Manikgog committed:
Приведение методов и документации в соответствие с примером API.

Manikgog committed:
Написал контроллер для комментариев CommentController. Исправил DTO, чтобы в swagger'е были видны комментарии к полям.

Commit:
создал DTO классы, написал 4 метода контроллера AdsController и аннотации к ним, сервис-заглушка,

Manikgog committed:
Изменены классы для авторизации.

Manikgog committed:
Изменены классы для авторизации.

ASanderD committed:
Написал 3 метода контроллера, доки и методы-заглушки в сервис

Manikgog committed:
Merge branch 'refs/heads/feature-vetchanin-ads' into develop

ASanderD committed:
Написал 3 метода контроллера, доки и методы-заглушки в сервис

Manikgog committed:
Merge branch 'refs/heads/feature_dubograev_ads' into develop

# Conflicts:
#	src/main/java/ru/skypro/homework/config/WebSecurityConfig.java
#	src/main/java/ru/skypro/homework/controller/AdsController.java
#	src/main/java/ru/skypro/homework/dto/Ad.java
#	src/main/java/ru/skypro/homework/dto/Ads.java
#	src/main/java/ru/skypro/homework/dto/Comment.java
#	src/main/java/ru/skypro/homework/dto/Comments.java
#	src/main/java/ru/skypro/homework/dto/CreateOrUpdateAd.java
#	src/main/java/ru/skypro/homework/dto/ExtendedAd.java
#	src/main/java/ru/skypro/homework/filter/BasicAuthCorsFilter.java
#	src/main/java/ru/skypro/homework/service/AdsService.java
#	src/main/java/ru/skypro/homework/service/impl/AdsServiceImpl.java

Manikgog committed:
Устранены замечания наставника.

dmitry-bizin commented on pom.xml:
```diff
@@ -52,6 +52,20 @@
+    <!-- https://mvnrepository.com/artifact/javax.servlet/javax.servlet-api -->
+    <dependency>
+      <groupId>javax.servlet</groupId>
+      <artifactId>javax.servlet-api</artifactId>
```
зависимость не нужна, будет конфликтовать, надо удалить

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -0,0 +1,77 @@
+@RestController
+@RequestMapping("/ads")
+public class AdsController {
+    private final AdsServiceImpl adsService;
```
```suggestion
    private final AdsService adsService;
```
инъекцию надо делать по интерфейсу, а не реализации

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -0,0 +1,77 @@
+public class AdsController {
+    private final AdsServiceImpl adsService;
+
+    public AdsController(AdsServiceImpl adsService) {
```
```suggestion
    public AdsController(AdsService adsService) {
```
инъекцию надо делать по интерфейсу, а не реализации

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -0,0 +1,77 @@
+        this.adsService = adsService;
+    }
+
+    @Operation(summary = "Получение всех объявлений", description = "", tags = {"Объявления"})
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "OK", content =
+            @Content(mediaType = "application/json", schema = @Schema(implementation = Ads.class)))})
```
нет никакого смысла описывать эндпоинты аннотациями Swagger UI, у вас уже есть спецификация, в которой всё это есть, дублировать не нужно. Сейчас подход design-first

dmitry-bizin commented on pom.xml:
```diff
@@ -52,6 +52,20 @@
+    </dependency>
+    <dependency>
+      <groupId>org.springdoc</groupId>
+      <artifactId>springdoc-openapi-starter-webmvc-ui</artifactId>
```
нет никакого смысла описывать эндпоинты аннотациями Swagger UI, у вас уже есть спецификация, в которой всё это есть, дублировать не нужно. Сейчас подход design-first

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/AdsController.java:
```diff
@@ -0,0 +1,77 @@
+    public ResponseEntity<Ads> getAds() {
+        return ResponseEntity.ok().body(new Ads());
+    }
+
```
не хватает эндпоинта обновления картинки объявления

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/Comment.java:
```diff
@@ -0,0 +1,23 @@
+    @Schema(description = "дата и время создания комментария в миллисекундах с 00:00:00 01.01.1970")
+    long createdAt;
+    @Schema(description = "id комментария")
+    int id;
```
pk согласно спецификации

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/Register.java:
```diff
@@ -1,14 +1,23 @@
-
-    private String username;
+    @Schema(minLength = 4, maxLength = 32, description = "логин")
+    private String userName;
```
```suggestion
    private String username;
```
см. спецификацию

Manikgog committed:
Cозданы сущности и репозитории.

Manikgog committed:
В CommentEntity в поле text commentDefinition заменено на VARCHAR(64)

Manikgog committed:
Добавлен класс для чтения и записи файлов с изображениями.

Commit:
Добавлен mapstruct, добавлен интерфейс-маппер для объявлений

Manikgog committed:
Созданы интерфейсы UpdateUserMapper и UserMapper.

Manikgog committed:
Добавбены зависимости в pom.xml

Manikgog committed:
Добавбены две зависимости от классов UserMapper и UpdateUserMapper в класс UserServiceImpl.

ASanderD committed:
Написан CommentMapper.

Manikgog committed:
В UserEntity поле password заменено с Integer на String

Manikgog committed:
Merge branch 'refs/heads/gogolin_make_mappers' into develop

Manikgog committed:
Merge branch 'refs/heads/added-mapperAds-vetchanin' into develop

Manikgog committed:
Классы UpdateUserMapper и UserMapper перемещены в пакет mapper и пакет mappers удалён.

Manikgog committed:
Добавлен маппер RegisterMapper из Register в UserEntity, и изменены характеристики полей UserEntity: email, firstName, lastName, phone, role, password на nullable = false. T.е. они не могут быть NULL.

Manikgog committed:
В классе UserEntity в поле email добавлена constraint unique.

dmitry-bizin commented on pom.xml:
```diff
@@ -52,6 +52,27 @@
+    <dependency>
+      <groupId>org.mapstruct</groupId>
+      <artifactId>mapstruct</artifactId>
+      <version>1.5.5.Final</version>
```
выше есть пропертя для версии mapstruct, вы либо её используйте либо удалите

dmitry-bizin replied:
выше есть пропертя для версии mapstruct, вы либо её используйте либо удалите

dmitry-bizin commented on pom.xml:
```diff
@@ -78,6 +99,33 @@
+        <artifactId>maven-compiler-plugin</artifactId>
+        <version>3.5.1</version>
+        <configuration>
+          <source>9</source>
+          <target>9</target>
```
почему здесь стоит версия языка 9, а выше 17?

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/Ads.java:
```diff
@@ -0,0 +1,38 @@
+    @JsonProperty("result")
+    List<Ad> results;
+
+    public Ads count(Integer count) {
+        this.count = count;
+        return this;
+    }
+
+    public Ads results(List<Ad> results) {
+        this.results = results;
+        return this;
+    }
```
это можно сделать через аннотацию Setter ломбока

dmitry-bizin commented on src/main/java/ru/skypro/homework/exceptions/ImageNotFoundException.java:
```diff
@@ -0,0 +1,16 @@
+
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+@ResponseStatus
```
какой?

dmitry-bizin commented on src/main/java/ru/skypro/homework/mapper/CommentMapper.java:
```diff
@@ -0,0 +1,15 @@
+public interface CommentMapper {
+
+    @Mapping(target = "author", source = "author.id")
+    @Mapping(target = "pk",source = "ad.id")
```
здесь id комментария именно
```suggestion
    @Mapping(target = "pk",source = "id")
```

dmitry-bizin replied:
где маппинг для authorFirstName?

dmitry-bizin commented on src/main/java/ru/skypro/homework/mapper/UpdateUserMapper.java:
```diff
@@ -0,0 +1,12 @@
+
+@Mapper(componentModel = MappingConstants.ComponentModel.SPRING)
+public interface UpdateUserMapper {
+    UpdateUser toUpdateUser(UserEntity source);
```
здесь не наоборот ли нужно?

dmitry-bizin commented on src/main/java/ru/skypro/homework/mapper/CommentMapper.java:
```diff
@@ -0,0 +1,15 @@
+import ru.skypro.homework.entitiy.CommentEntity;
+
+
+@Mapper(componentModel = "spring")
```
```suggestion
@Mapper(componentModel = MappingConstants.ComponentModel.SPRING)
```

dmitry-bizin commented on src/main/java/ru/skypro/homework/mapper/AdMapper.java:
```diff
@@ -0,0 +1,22 @@
+import ru.skypro.homework.dto.ExtendedAd;
+import ru.skypro.homework.entitiy.AdEntity;
+
+@Mapper(componentModel = "spring",uses = AdEntity.class)
```
```suggestion
@Mapper(componentModel = MappingConstants.ComponentModel.SPRING, uses = AdEntity.class)
```

dmitry-bizin replied:
uses не нужно, там мапперы указывают, а у вас просто класс

dmitry-bizin commented on src/main/resources/application.properties:
```diff
@@ -0,0 +1,8 @@
+spring.datasource.url=jdbc:postgresql://localhost:5432/resaledb
+spring.datasource.username=${USER_NAME}
+spring.datasource.password=${DATASOURCE_PASSWORD}
+spring.jpa.hibernate.ddl-auto=validate
```
а где миграции для этой стратегии? откуда у вас таблицы возьмутся?

Manikgog committed:
Замечания от наставника устранены. Созданы файлы для создания таблиц с помощью liquibase.

ASanderD committed:
Написан liquibase, внесены некоторые изменения с учетом замечаний преподователя.

Manikgog committed:
Добавлено ограничение в поле email user_table - unique.

Manikgog committed:
Написал security и несколько методов UserController.

Manikgog committed:
Написал security и несколько методов AdsController.

Manikgog committed:
Написал все методы AdsService и UserServiceImpl.

Manikgog committed:
Сделал разграничение для обращения к методам для работы с методами AdsController, в зависимости от роли пользователя и того является ли он автором объявления.

Manikgog committed:
Сделал логирование в классах: AdsController, AdsServiceImpl, AuthController, AuthServiceImpl, CheckAccessService, CommentController, FileManager, UserController, UserServiceImpl.

ASanderD committed:
Написаны контроллер и сервис для работы с комментариями.

Manikgog committed:
Merge remote-tracking branch 'refs/remotes/origin/feature_dubograev_commentService' into develop

# Conflicts:
#	pom.xml
#	src/main/java/ru/skypro/homework/controller/CommentController.java
#	src/main/java/ru/skypro/homework/exceptions/ImageNotFoundException.java
#	src/main/java/ru/skypro/homework/mapper/AdMapper.java
#	src/main/java/ru/skypro/homework/mapper/CommentMapper.java
#	src/main/resources/application.properties
#	src/main/resources/liquibase/v_1/create-ad_table.xml
#	src/main/resources/liquibase/v_1/create-comment_table.xml
#	src/main/resources/liquibase/v_1/create-user_table.xml

ASanderD committed:
Переписаны Comment контроллер и сервис с учетом проверки авторизации.

Manikgog committed:
Исправлены недочёты

Manikgog committed:
Merge branch 'refs/heads/dubograev_security_feature' into develop

# Conflicts:
#	src/main/java/ru/skypro/homework/check/CheckAccessService.java
#	src/main/java/ru/skypro/homework/repository/CommentRepo.java
#	src/main/java/ru/skypro/homework/service/impl/CommentServiceImpl.java

Manikgog committed:
Удален exception CommentEntityException.

Manikgog committed:
Исправлена ошибка в методе isAdminOrOwnerComment в строке 42 вместо || вставлено &&, потому что пользователь должен быть и владельцем объявления и у объявления должен быть указанный комментарий.

Manikgog committed:
Ошибка несоответствия комментария объявлению не относится к авторизации, поэтому проверка соответствия комментария объвлению вынесена отдельно в методе isAdminOrOwnerComment класса CheckAccessService (строка 41). Добавбено исключение CommentDoesNotMatchTheAdException.

dmitry-bizin commented on src/main/java/ru/skypro/homework/check/CheckAccessService.java:
```diff
@@ -0,0 +1,81 @@
+        if(!adEntity.equals(commentEntity.getAd())){
+            throw new CommentDoesNotMatchTheAdException("The comment does not match the ad");
+        }
+        return userEntity.getRole().name().equals("ADMIN")
```
лучше сравнивать enum-ы, а не строки

dmitry-bizin commented on src/main/java/ru/skypro/homework/check/CheckAccessService.java:
```diff
@@ -0,0 +1,81 @@
+                    log.error("An EntityNotFoundException " + "(Ad " + adId + " not found)" + "exception was thrown when calling the isAdminOrOwnerAd method of CheckAccessService");
+                    return new EntityNotFoundException("Ad " + adId + " not found");
+                });
+        return userEntity.getRole().name().equals("ADMIN")
```
лучше сравнивать enum-ы, а не строки

dmitry-bizin commented on src/main/java/ru/skypro/homework/check/CheckAccessService.java:
```diff
@@ -0,0 +1,81 @@
+                    log.error("An EntityNotFoundException " + "(Ad " + id + " not found)" + "exception was thrown when calling the isAdminOrOwnerAd method of CheckAccessService");
+                    return new EntityNotFoundException("Ad " + id + " not found");
+                });
+        return userEntity.getRole().name().equals("ADMIN")
+                || userEntity.getRole().name().equals("USER")
```
какой смысл в этих проверках? ролей и так всего 2 штуки, поэтому логично, что роль будет либо юзер, либо админ, зачем это проверять?

dmitry-bizin commented on src/main/java/ru/skypro/homework/check/CheckAccessService.java:
```diff
@@ -0,0 +1,81 @@
+                });
+        return userEntity.getRole().name().equals("ADMIN")
+                || userEntity.getRole().name().equals("USER")
+                || userEntity.getId().equals(adEntity.getAuthor().getId());
```
эту проверку я также не понял зачем делать, если по заданию надо ещё проверять на админа и у вас выше есть эти методы уже

dmitry-bizin commented on src/main/java/ru/skypro/homework/check/CheckService.java:
```diff
@@ -0,0 +1,48 @@
+@Component
+public class CheckService {
+
+    public void checkString(int minLength, int maxLength, String string) {
```
все эти проверки можно было заменить на аннотации validation

dmitry-bizin commented on src/main/java/ru/skypro/homework/check/CheckService.java:
```diff
@@ -0,0 +1,48 @@
+                return;
+            }
+        }
+        throw new WrongRoleException("Указана несуществующая роль");
```
а как такое может быть? только если null, по-другому никак, потому что компилятор контролирует, что значение enum будет какое-то из того, которое внутри описано, проверка смысла не имеет

dmitry-bizin commented on src/main/java/ru/skypro/homework/check/CheckService.java:
```diff
@@ -0,0 +1,48 @@
+    }
+
+    public void checkPhone(String pattern, String phone){
+        Pattern pat = Pattern.compile(pattern);
```
паттерн лучше сделать полем, чтобы не создавать каждый раз заново

dmitry-bizin commented on src/main/java/ru/skypro/homework/config/MyUserDetails.java:
```diff
@@ -0,0 +1,58 @@
+
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        return List.of(new SimpleGrantedAuthority(user.getRole().name()));
```
вот здесь косяк, авторити - это не роль, и должна начинаться с ROLE_, т.е, ROLE_USER и ROLE_ADMIN должно быть

dmitry-bizin commented on src/main/java/ru/skypro/homework/config/MyUserDetailsService.java:
```diff
@@ -0,0 +1,22 @@
+    @Override
+    public MyUserDetails loadUserByUsername(String username) {
+        UserEntity user = userRepo.findByEmail(username).orElseThrow(()-> new EntityNotFoundException("User is not found"));
+        return new MyUserDetails(user);
```
можно было не создавать свой класс, а использовать готовый User, у которого даже билдер есть

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentController.java:
```diff
@@ -28,16 +37,18 @@ public class CommentController {
+                    description = "Not found",
+                    content = @Content())
     })
+    @PreAuthorize("@checkAccessService.isAuthorizedUser(#id, authentication)")
```
здесь не нужна проверка, нужна только для удаления и обновления

dmitry-bizin commented on src/main/java/ru/skypro/homework/controller/CommentController.java:
```diff
@@ -51,13 +62,16 @@ public ResponseEntity<Comments> get(@PathVariable Integer id){
                     description = "Not found",
                     content = @Content())
     })
+    @PreAuthorize("@checkAccessService.isAuthorizedUser(#id, authentication)")
```
здесь не нужна проверка, нужна только для удаления и обновления

dmitry-bizin replied:
любой пользователь может оставить комментарий под любым объявлением, а не только создатель объявления или админ

dmitry-bizin commented on src/main/java/ru/skypro/homework/mapper/AdEntityToExtendedAdMapper.java:
```diff
@@ -0,0 +1,24 @@
+    public ExtendedAd perform(AdEntity adEntity) {
+        ExtendedAd extendedAd = new ExtendedAd();
+        UserEntity userEntity = adEntity.getAuthor();
+        extendedAd.setPk(Math.toIntExact(adEntity.getId()));
```
зачем это нужно? почему бы просто не сделать одинаковые типы?

dmitry-bizin commented on src/main/java/ru/skypro/homework/mapper/AdMapper.java:
```diff
@@ -19,4 +21,6 @@ public interface AdMapper {
     @Mapping(target = "phone", source = "author.phone")
     ExtendedAd adEntityToExtendedAd(AdEntity source);
 
+    void CreateOrUpdateAdToAdEntity(CreateOrUpdateAd source, @MappingTarget AdEntity target);
```
названия методов с маленькой буквы надо писать

dmitry-bizin commented on src/main/java/ru/skypro/homework/mapper/CommentMapper.java:
```diff
@@ -2,14 +2,17 @@
+    @Mapping(target = "pk",source = "id")
+    @Mapping(target = "authorImage", source = "author.image")
+    @Mapping(target = "authorFirstName",source = "author.firstName")
     Comment CommentEntityToComment(CommentEntity commentEntity);
```
названия методов с маленькой буквы надо писать

dmitry-bizin commented on src/main/java/ru/skypro/homework/mapper/RegisterMapper.java:
```diff
@@ -2,11 +2,10 @@
+import ru.skypro.homework.entity.UserEntity;
 
-@Mapper(componentModel = MappingConstants.ComponentModel.SPRING)
+@Mapper(componentModel = "spring")
```
```suggestion
@Mapper(componentModel = MappingConstants.ComponentModel.SPRING)
```

dmitry-bizin commented on src/main/java/ru/skypro/homework/mapper/UpdateUserMapper.java:
```diff
@@ -1,12 +1,13 @@
+import ru.skypro.homework.entity.UserEntity;
 
-@Mapper(componentModel = MappingConstants.ComponentModel.SPRING)
+@Mapper(componentModel = "spring")
```
```suggestion
@Mapper(componentModel = MappingConstants.ComponentModel.SPRING)
```

dmitry-bizin commented on src/main/java/ru/skypro/homework/mapper/UserMapper.java:
```diff
@@ -1,11 +1,10 @@
+import ru.skypro.homework.entity.UserEntity;
 
-@Mapper(componentModel = MappingConstants.ComponentModel.SPRING)
+@Mapper(componentModel = "spring")
```
```suggestion
@Mapper(componentModel = MappingConstants.ComponentModel.SPRING)
```

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/AdsServiceImpl.java:
```diff
@@ -1,54 +1,168 @@
+    private final CheckService checkService;
+    private final Constants constants;
+    private final AdEntityToExtendedAdMapper adEntityToExtendedAdMapper = new AdEntityToExtendedAdMapper();
+    private final AdToAdEntity adToAdEntity = new AdToAdEntity();
```
а это зачем?

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/AdsServiceImpl.java:
```diff
@@ -1,54 +1,168 @@
+    private final FileManager fileManager;
+    private final CheckService checkService;
+    private final Constants constants;
+    private final AdEntityToExtendedAdMapper adEntityToExtendedAdMapper = new AdEntityToExtendedAdMapper();
```
руками ничего создавать не нужно, инъекцию надо делать

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/AuthServiceImpl.java:
```diff
@@ -1,46 +1,78 @@
 public class AuthServiceImpl implements AuthService {
 
-    private final UserDetailsManager manager;
+    private final MyUserDetailsService myUserDetailService;
```
```suggestion
    private final UserDetailsService myUserDetailService;
```
инъекцию по интерфейсу делать нужно

dmitry-bizin commented on src/main/java/ru/skypro/homework/service/impl/CommentServiceImpl.java:
```diff
@@ -0,0 +1,101 @@
+     */
+    @Override
+    public Comment update(Integer adId, Integer commentId, CreateOrUpdateComment newComment) {
+        CommentEntity commentEntity = commentRepo.findById((long) commentId).orElseThrow();
```
зачем делать каст, если можно просто сделать одного типа параметры?

Manikgog committed:
Исправлены ошибки наставника.

Manikgog committed:
Исправление ошибок наставника.

Manikgog committed:
Исправление ошибок  указанных наставником.

Manikgog committed:
Исправление ошибок  указанных наставником.

Manikgog committed:
Сделал отображение фотографии пользователя на фронтенде.

dmitry-bizin commented on pom.xml:
```diff
@@ -72,8 +72,15 @@
+      <dependency>
+          <groupId>org.liquibase</groupId>
+          <artifactId>liquibase-core</artifactId>
+          <version>4.28.0</version>
```
версию писать не нужно

dmitry-bizin commented on src/main/java/ru/skypro/homework/check/CheckAccessService.java:
```diff
@@ -41,7 +42,7 @@ public boolean isAdminOrOwnerComment(Long adId, Long commentId, Authentication a
             throw new CommentDoesNotMatchTheAdException("The comment does not match the ad");
         }
-        return userEntity.getRole().name().equals("ADMIN")
+        return userEntity.getRole().equals(Role.ROLE_ADMIN)
```
enum-ы можно через == сравнивать

dmitry-bizin commented on src/main/java/ru/skypro/homework/check/CheckAccessService.java:
```diff
@@ -58,24 +59,8 @@ public boolean isAdminOrOwnerAd(Long adId, Authentication authentication) {
                     return new EntityNotFoundException("Ad " + adId + " not found");
                 });
-        return userEntity.getRole().name().equals("ADMIN")
+        return userEntity.getRole().equals(Role.ROLE_ADMIN)
```
enum-ы можно через == сравнивать

dmitry-bizin commented on src/main/java/ru/skypro/homework/dto/Role.java:
```diff
@@ -1,5 +1,5 @@
 
 public enum Role {
-    USER, ADMIN
+    ROLE_USER, ROLE_ADMIN
```
а зачем ROLE_ в начале? опять напутали) роли - USER и ADMIN, вас фронт теперь понимать не будет и для админа не будет показывать крестики, потому что фронт понимает роль ADMIN, а не ROLE_ADMIN. А вот авторити - ROLE_USER и ROLE_ADMIN

Manikgog committed:
Сделал отображение товара на фронтетде

Manikgog committed:
1. Сделал, чтобы если фотография пользователя один раз появилась, то название файла не меняется, происходит перезапись файла с тем же названием.
2. Сделал, чтобы фотографии отображались в комментариях.
3. Сделал доступными для неатутентифицированных пользователей просмотр объявлений.

Manikgog committed:
Сделал, чтобы если фотография объявления один раз появилась, то название файла не меняется, происходит перезапись файла с тем же названием.

Manikgog committed:
Начал писать класс для интеграционного тестирования UserControllerTestRestTemplateTest.

Manikgog committed:
Сделал, чтобы у неаутентифицированного пользователя показывались объявления с фотографиями.

Manikgog committed:
Написан метод setPassword_positiveTest().

Manikgog committed:
Написан метод setPassword_negativeTest(), updateUser_positiveTest(), updateImage_positiveTest(), getUser_positiveTest().

Manikgog committed:
Написаны методы: getImage_positiveTest(), uploadImage_positiveTest(). В методе afterEach() добавлено удаление всех файлов изображений, названия которых были записаны в базу данных, перед её очисткой.

Manikgog committed:
Исправлена ошибка при вставке фото пользователя при изначальном отсутствии фото в базе данных выбрасывалось NPE. Также исправлен метод удаления объявления - перед удалением объявления удаляются все комментарии к объявлению.

Manikgog committed:
Merge branch 'refs/heads/gogolin_user_controller_test' into develop

# Conflicts:
#	src/main/java/ru/skypro/homework/config/WebSecurityConfig.java

Manikgog committed:
Исправлены мелкие недочёты в UserControllerTestRestTemplateTest и начато написание AuthControllerTestRestTemplate.

Manikgog committed:
Написал тесты для всех мапперов.

Commit:
Tests AdsController

Manikgog committed:
Merge remote-tracking branch 'refs/remotes/origin/AdsControllerTest-Vetchanin' into develop

# Conflicts:
#	test_ad_images/2.jpg

Manikgog committed:
Разбирался с тестированием методов uploadImage и getImage класса UserController.

Manikgog committed:
Merge remote-tracking branch 'refs/remotes/origin/dubograev_tests_feature' into develop

Manikgog committed:
2.jpg

ASanderD committed:
Написал CommentControllerTestRestTemplateTest.

ASanderD committed:
Написал CommentControllerTestRestTemplateTest.

Manikgog committed:
Устранены ошибки, которые возникали при тестировании при сборке проекта maven и написан файл README.md.

Manikgog committed:
В классе AdsControllerTestRestTemplateTest добавлено удаление файлов с фотографиями после каждого теста.

Manikgog committed:
папка тест приведена к виду папки java

Manikgog committed:
Сделано создание папок для пользователей и для объявлений при запуске программы. Удаление файлов, которые не содержатся в базе данных при запуске и при изменении изображения пользователя.

Manikgog committed:
Исправлены тесты, которые выдавали ошибку -parameters в контроллеры добавлены @PathVariable("id")

Manikgog committed:
Исправлены тесты с учётом того, что при запуске приложения стираются файлы, которых нет в базе данных. В методах с аннотацией @BeforeEach добавлено создание файла jpg

Manikgog committed:
Merge branch 'refs/heads/gogolin_create_folders' into develop

# Conflicts:
#	src/main/java/ru/skypro/homework/controller/CommentController.java
#	src/test/java/ru/skypro/homework/controller/AdsControllerTestRestTemplateTest.java

Manikgog committed:
Добавил имена, участвующих в разработке приложения.

Manikgog committed:
Добавил комментарий для метода prepare() класса FileManager

Manikgog committed:
Update README.md

Manikgog committed:
Update README.md

Manikgog committed:
Merge branch 'main' into develop
